File: dotnet/Middleware.Contracts/Events/AppraisalUnderwriterAssignedEvent.cs
namespace Middleware.Contracts.Events;

// UC4 · Appraisal Documents — StatusCode=6 workflow completed; UW assigned
public sealed class AppraisalUnderwriterAssignedEvent
{
    public string AppraisalId { get; set; } = string.Empty;
    public string PolicyNumber { get; set; } = string.Empty;
    public string InspectionId { get; set; } = string.Empty;
    public string UWAssignmentType { get; set; } = string.Empty;
    public string AssignedTo { get; set; } = string.Empty;
    public int SuspenseDays { get; set; }
    public string PLUWWorkItemId { get; set; } = string.Empty;
    public string CorrelationId { get; set; } = string.Empty;
    public DateTimeOffset CompletedAt { get; set; }
}

File: java/common/src/main/java/com/ais/middleware/common/events/prs/AppraisalUnderwriterAssignedEvent.java
package com.ais.middleware.common.events.prs;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.time.OffsetDateTime;

/**
 * Published when the AppraisalReceivedSaga completes StatusCode=6 (UW Determination) flow.
 * Signals that an underwriter has been assigned and downstream systems have been updated.
 *
 * Topic: prs.events.appraisal-uw-assigned
 */
public record AppraisalUnderwriterAssignedEvent(
        String correlationId,
        String appraisalId,
        String policyNumber,
        String uwAssignment,
        int suspenseDays,
        String pluwReferenceId,
        OffsetDateTime assignedAt
) {
    @JsonCreator
    public AppraisalUnderwriterAssignedEvent(
            @JsonProperty("correlationId") String correlationId,
            @JsonProperty("appraisalId") String appraisalId,
            @JsonProperty("policyNumber") String policyNumber,
            @JsonProperty("uwAssignment") String uwAssignment,
            @JsonProperty("suspenseDays") int suspenseDays,
            @JsonProperty("pluwReferenceId") String pluwReferenceId,
            @JsonProperty("assignedAt") OffsetDateTime assignedAt
    ) {
        this.correlationId = correlationId;
        this.appraisalId = appraisalId;
        this.policyNumber = policyNumber;
        this.uwAssignment = uwAssignment;
        this.suspenseDays = suspenseDays;
        this.pluwReferenceId = pluwReferenceId;
        this.assignedAt = assignedAt;
    }
}
